pypeit/specobjs.py

```diff
@@ -1074,5 +1083,15 @@ def lst_to_array(lst, mask=None):
     if isinstance(lst[0], units.Quantity):
         return units.Quantity(lst)[mask]
     else:
-        return np.array(lst)[mask]
+        # if all the elements of lst have the same type, np.array(lst)[mask] will work
+        if len(set(map(type, lst))) == 1:
+            return np.array(lst)[mask]
+        else:
+            # if not, we need to use dtype="object"
+            return np.array(lst, dtype="object")[mask]
+        # TODO: The dtype="object" is needed for the case where one element of lst is not a list but None.
```

Comment:
good to have @kbwestfall 's eyes on this..

Reply:
I think this change is fine, but I'd have a full picture of why we need it.  I.e., this ValueError must not have been something that is often tripped in the code.  What cases cause it to be tripped?  And are these cases that we must support, or are we just trying to keep the code from faulting here, even though the root cause is actually something at we should avoid?

Reply:
@kbwestfall the error "_ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions.._." happens pretty often, especially for echelle, but not only. I have found it happens when trying to unpack SpecObjs. For example, if we are trying to get all the OPT fluxes from a specific SpecObjs, and for one slit/order the OPT extraction failed (but not the BOX extraction) and therefore `OPT_COUNTS` is `None`,  the list `lst` in this method would be something like `[array, array, array, None, array]`, which makes np.array to fail and give the ValueError. With `dtype='object'` this error does not occur, but some functions later on in the code do not like to work with object arrays. I thought this was a compromise to deal with this situation.

I tried to fix the root cause partially in hires, by forcing standard star echelle data to have maximum one object per order and other science echelle data maximum 2 objects per order, since most of the failed OPT extraction are due to sky subtraction problems (caused by spurious detections). But anyway, I think we need a way to deal with this in general.